cactusbot/cached/caching.py

```diff
@@ -1,18 +1,32 @@
+import json
+import time
+
+
 class CacheUtils(object):
 
     # You have found MysticalMage's code, turn back now, no good lies ahead
-    def in_cache_followers(user):
-        """ Returns if user is in the cache. """
-        follower_cache = open("caches/CacheFollowers.txt", "r")
-        for line in follower_cache:
-            if user in line:
-                follower_cache.close()
-                return True
-        follower_cache.close()
+    def __init__(self, cacheName):
+        super().__init__()
+
+        self.cacheName = cacheName
+
+    def in_cache(self, user):
+        """ Returns if user is in the cache, and if so, the time in secs """
+        with open(self.cacheName) as cache:
+            cachedata = json.load(cache)
+        if user in cachedata.keys():
+            return True
         return False
 
-    def cache_followers_add(user):
+    def cache_add(self, user):
         """ Adds user to cache with newline at end. """
-        follower_cache = open("caches/CacheFollowers.txt", "a+")
-        follower_cache.write(user + "\n")
-        follower_cache.close()
+        with open(self.cacheName, "r") as cache:
+            cachedata = json.load(cache)
+        cachedata[user] = time.gmtime()
```

Comment:
`time.gmtime()` is going to be hard to work with. Maybe use seconds since epoch?
```python
time.mktime(time.gmtime())  # 1476173154.0
```